src/bytes/bytes_lemmas.gobra

```diff
@@ -0,0 +1,505 @@
+package bytes
+
+//+gobra
+
+import (
+	. "verification/utils/definitions"
+	sl "gobra-libs/byteslice"
+	bytes "gobra-libs/bytes"
+	. "bytes/spec"
+	"unicode/utf8"
+)
+
+decreases
+requires acc(sl.Bytes(b, 0, len(b)), _)
+pure func String(b []byte) string  {
+	return unfolding acc(sl.Bytes(b, 0, len(b)), _) in string(b)
+}
+
+// We cannot yet prove this with the current incomplete encoding of strings
+ghost
+requires acc(sl.Bytes(a, 0, len(a)), _)
+requires acc(sl.Bytes(b, 0, len(b)), _)
+ensures (String(a) == String(b)) == (sl.View(a) == sl.View(b))
+decreases
+pure func stringEqualsImplViewEquals(a, b []byte) bool
+
+
+// Count
+
+/*
+
+ghost
+decreases len(s)
+pure func SpecCount(s, sep seq[byte]) (res int) {
+	return len(sep) == 0 ?
+		len(utf8.Codepoints(s)) + 1 :
+		len(s) < len(sep) ?
+			0 :
+			s[:len(sep)] == sep ?
+				1 + SpecCount(s[len(sep):], sep) :
+				SpecCount(s[1:], sep)
+}
+
+*/
+
+ghost
+requires len(sep) > 0
+requires 0 <= i
+ensures forall j int :: {j in res} j in res ==> InRangeInc(j, i, len(s)-len(sep)) && s[j:j+len(sep)] == sep
+ensures forall j, k int :: {j in res, k in res} j in res && k in res && j < k ==> j + len(sep) <= k
+decreases len(s) - i
+pure func SpecCountIndices(s, sep seq[byte], i int) (res set[int]) {
+	return i >= len(s) ?
+		set[int]{} :
+		s[i:i+len(sep)] == sep ?
+			set[int]{i} union SpecCountIndices(s, sep, i + len(sep)) :
+			SpecCountIndices(s, sep, i + 1)
+}
+
+ghost
+requires forall j int :: {j in indices} j in indices ==> 0 <= j && j < len(b)
+requires forall j int :: {j in indices} j in indices ==> ( b[j] == c)
+requires forall j int :: {j in indices} !(j in indices) ==> ( j < 0 || len(b) <= j || b[j] != c)
+ensures indices == SpecCountIndices(b, seq[byte]{c}, 0)
+decreases
+func lemmaBytealgCount(indices set[int], b seq[byte], c byte) {
+	TODO() // before implementing this lemma, we should consider changing the specs of bytealg.Count to use SpecCountIndices
+}
```

Comment:
please mark it as `trusted` and drop the body, otherwise it becomes very hard to find this assumption